WireAPI/Sources/WireAPI/Authentication/PersistentAuthenticationStorage.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireFoundation
+
+public actor PersistentAuthenticationStorage: AuthenticationStorage {
+
+    private typealias Error = PersistentAuthenticationStorageError
+
+    private let userID: UUID
+    private var accessToken: AccessToken?
+    private let sharedUserDefaults: UserDefaults
+    private let keychain: any KeychainProtocol
+    private static let cookieEncryptionKeyKey = "ZMCookieKey"
+
+    public init(
+        userID: UUID,
+        sharedUserDefaults: UserDefaults,
+        keychain: any KeychainProtocol
+    ) {
+        self.userID = userID
+        self.sharedUserDefaults = sharedUserDefaults
+        self.keychain = keychain
+    }
+
+    // MARK: - Access token
+
+    public func storeAccessToken(_ accessToken: AccessToken) async {
+        self.accessToken = accessToken
+    }
+
+    public func fetchAccessToken() async -> AccessToken? {
+        accessToken
+    }
+
+    // MARK: - Cookie
+
+    public func storeCookies(_ cookies: [HTTPCookie]) async throws {
+        let cookieData = try HTTPCookieCodec.encodeCookies(cookies)
+        try await storeCookieData(cookieData)
+    }
+
+    public func fetchCookies() async throws -> [HTTPCookie] {
```

Comment:
If the app is deleted and reinstalled and there is old cookie data from the previous install which **hopefully** cannot be decrypted, does this method throw an error?

If yes, perhaps we should consider if this is really an error. Alternatively it should be documented that this kind of error maybe expected as a doc comment so the consumer knows to relax.

Reply:
done 65b4f58aa1a23e8f1aa3142d6981f4fac153a204